pkg/solana/monitor/balance.go

```diff
@@ -26,34 +28,33 @@ type BalanceClient interface {
 }
 
 // NewBalanceMonitor returns a balance monitoring services.Service which reports the SOL balance of all ks keys to prometheus.
-func NewBalanceMonitor(chainID string, cfg Config, lggr logger.Logger, ks Keystore, newReader func() (BalanceClient, error)) services.Service {
-	return newBalanceMonitor(chainID, cfg, lggr, ks, newReader)
+func NewBalanceMonitor(chainID string, cfg Config, lggr logger.Logger, ks Keystore, reader internal.Loader[BalanceClient]) services.Service {
```

Comment:
why these changes?

Reply:
BalanceMonitor was also doing caching of a single RPC which this allows us to disable. It also had its own lazy loading logic which should instead use the same lazyloader as everything else.